logging/google/cloud/logging/handlers/_helpers.py

```diff
@@ -37,3 +47,59 @@ def format_stackdriver_json(record, message):
     }
 
     return json.dumps(payload)
+
+
+def get_trace_id_from_flask():
+    """Get trace_id from flask request headers.
+
+    :rtype: str
+    :return: Trace_id in HTTP request headers.
+    """
+    if not flask or not flask.request:
+        return None
+
+    header = flask.request.headers.get(_FLASK_TRACE_HEADER)
+
+    if not header:
+        return None
+
+    trace_id = header.split('/')[0]
+
+    return trace_id
+
+
+def get_trace_id_from_django():
+    """Get trace_id from django request headers.
+
+    :rtype: str
+    :return: Trace_id in HTTP request headers.
+    """
+    request = _get_request()
+
+    if not request:
+        return None
+
+    try:
+        header = request.META[_DJANGO_TRACE_HEADER]
```

Comment:
An interesting behavioral difference between flask and django here. The flask one will return `None` if the trace header is set but is an empty string, whereas the django one will return an empty string. I think you mentioned empty strings cause issues with the logging API? The django one should likely use `request.META.get(...)` instead of this to match the behavior of the flask one.